### PR TITLE
Subscription: java client supports Create Topic If Not Exists and Drop Topic If Exists

### PR DESCRIPTION
**Pull request by luoluoyuyu**
## Description
As Title

<hr>

This PR has:
- [ ] been self-reviewed.
    - [ ] concurrent read
    - [ ] concurrent write
    - [ ] concurrent read and write 
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. 
- [ ] added or updated version, __license__, or notice information
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious 
  for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold 
  for code coverage.
- [ ] added integration tests.
- [ ] been tested in a test IoTDB cluster.

<hr>

### CODE REVIEW

**Comment by luoluoyuyu** [comment]
> Can the existing createTopic and dropTopic implementations be removed and reuse the newly added code directly?

I mainly considered compatibility. If existing users use the original function, there may be problems, so I added a new function

File: iotdb-client/session/src/main/java/org/apache/iotdb/session/subscription/SubscriptionSession.java [hunk 1]
```diff
@@ -110,12 +110,49 @@ public void createTopic(final String topicName, final Properties properties)
     executeNonQueryStatement(sql);
   }
 
+  public void createTopic(
```
Comment (SteveYurongSu) [inline iotdb-client/session/src/main/java/org/apache/iotdb/session/subscription/SubscriptionSession.java:113]:
```suggestion
  public void createTopicIfNotExists(
```

File: iotdb-client/session/src/main/java/org/apache/iotdb/session/subscription/SubscriptionSession.java [hunk 2]
```diff
@@ -110,12 +110,49 @@ public void createTopic(final String topicName, final Properties properties)
...
+    executeNonQueryStatement(sql);
+  }
+
   public void dropTopic(final String topicName)
       throws IoTDBConnectionException, StatementExecutionException {
     final String sql = String.format("DROP TOPIC %s", topicName);
     executeNonQueryStatement(sql);
   }
 
+  public void dropTopic(final String topicName, final boolean isSetIfExistsCondition)
```
Comment (SteveYurongSu) [inline iotdb-client/session/src/main/java/org/apache/iotdb/session/subscription/SubscriptionSession.java:147]:
```suggestion
  public void dropTopicIfExists(final String topicName, final boolean isSetIfExistsCondition)
```

File: iotdb-client/session/src/main/java/org/apache/iotdb/session/subscription/SubscriptionSession.java (hunk 1 above)
Comment (SteveYurongSu) [reply iotdb-client/session/src/main/java/org/apache/iotdb/session/subscription/SubscriptionSession.java:113]: Add 2 new createTopicIfNotExists methods for 2 createTopic methods

**Comment by VGalaxies** [comment]
> > Can the existing createTopic and dropTopic implementations be removed and reuse the newly added code directly?
> 
> I mainly considered compatibility. If existing users use the original function, there may be problems, so I added a new function

I feel that for this reuse, there is no change in the SDK interface and semantics, so there should be no problem... cc @SteveYurongSu

**Review by SteveYurongSu — APPROVED** [review]
Lovely!

**Pull request merged**